src/EditorFeatures/CSharpTest/Diagnostics/UseImplicitOrExplicitType/UseExplicitTypeTests.cs

```diff
@@ -256,6 +255,154 @@ void Method()
 }", new TestParameters(options: ExplicitTypeEverywhere()));
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExplicitType)]
+        [WorkItem(23752, "https://github.com/dotnet/roslyn/issues/23752")]
+        public async Task OnDeconstructionVar()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+class Program
+{
+    void M()
+    {
+        [|var|] (x, y) = new Program();
+    }
+    void Deconstruct(out int i, out string s) { i = 1; s = ""hello""; }
+}", @"using System;
+class Program
+{
+    void M()
+    {
+        (int x, string y) = new Program();
+    }
+    void Deconstruct(out int i, out string s) { i = 1; s = ""hello""; }
+}", options: ExplicitTypeEverywhere());
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExplicitType)]
+        [WorkItem(23752, "https://github.com/dotnet/roslyn/issues/23752")]
+        public async Task OnNestedDeconstructionVar()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+class Program
+{
+    void M()
+    {
+        [|var|] (x, (y, z)) = new Program();
+    }
+    void Deconstruct(out int i, out Program s) { i = 1; s = null; }
+}", @"using System;
+class Program
+{
+    void M()
+    {
+        (int x, (int y, Program z)) = new Program();
+    }
+    void Deconstruct(out int i, out Program s) { i = 1; s = null; }
+}", options: ExplicitTypeEverywhere());
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExplicitType)]
+        [WorkItem(23752, "https://github.com/dotnet/roslyn/issues/23752")]
+        public async Task OnForeachNestedDeconstructionVar()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+class Program
+{
+    void M()
+    {
+        foreach ([|var|] (x, (y, z)) in new[] { new Program() } { }
+    }
+    void Deconstruct(out int i, out Program s) { i = 1; s = null; }
+}", @"using System;
+class Program
+{
+    void M()
+    {
+        foreach ((int x, (int y, Program z)) in new[] { new Program() } { }
+    }
+    void Deconstruct(out int i, out Program s) { i = 1; s = null; }
+}", options: ExplicitTypeEverywhere());
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExplicitType)]
+        [WorkItem(23752, "https://github.com/dotnet/roslyn/issues/23752")]
+        public async Task OnNestedDeconstructionVarWithTrivia()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+class Program
+{
+    void M()
+    {
+        /*before*/[|var|]/*lost*/ (/*x1*/x/*x2*/, /*yz1*/(/*y1*/y/*y2*/, /*z1*/z/*z2*/)/*yz2*/) /*after*/ = new Program();
+    }
+    void Deconstruct(out int i, out Program s) { i = 1; s = null; }
+}", @"using System;
+class Program
+{
+    void M()
+    {
+        /*before*/(
+/*x1*/int x/*x2*/,
+/*yz1*/(
+/*y1*/int y/*y2*/,
+/*z1*/Program z/*z2*/)/*yz2*/) /*after*/ = new Program();
```

Comment:
I'm not sure what causes this weird formatting. I'll check with @DustinCampbell for any tips before debugging in depth.

Reply:
Thanks @DustinCampbell for the [pointer](https://github.com/dotnet/roslyn/blob/master/src/Workspaces/CSharp/Portable/Simplification/Reducers/CSharpParenthesesReducer.cs#L33-L58).
If I understood properly, the approach is to drop some trivia that would cause formatting problems. I'm not sure I like this approach for this PR.
I'd rather either: (1) leave the odd formatting for the edge case of comments in such positions in a deconstruction, or (2) fix the formatter (it should not introduce newlines in certain positions).

If the IDE team is ok, I'll leave the current behavior (odd formatting).

Reply:
This looks pretty broken if you ask me...

Reply:
@jasonmalinowski I spent the afternoon deep in trivia formatter and could not figure it out. 
As far as I can tell the root cause of the newlines getting added is `AbstractTriviaFormatter.GetLineColumnOfWhitespace` which returns a value with lines=1. This causes `AbstractTriviaFormatter.AddWhitespaceTextChange` to add a newline.

I filed a follow-up issue: https://github.com/dotnet/roslyn/issues/24060

Reply:
Never mind. I finally understood how Dustin's workaround works and applied a similar trick ;-)

The key is that parens and commas introduced by the codefix come with elastic whitespace trivia which causes the formatter to misbehave. Removing those avoids the problem.

Reply:
@jasonmalinowski Any other review feedback?